qcfractal/interface/collections/collection.py

```diff
@@ -77,6 +77,8 @@ class DataModel(ProtoModel):
         view_metadata: Optional[Dict[str, str]] = None
         view_available: bool = False
 
+        metadata: Optional[Dict[str, Any]] = {}
```

Comment:
We could consider growing this over time `Union[MlMetaData, Dict[str, Any], None]`. As long as there is a concrete key in the models we can automatically cast them to the correct ProtoModel.

A key could be something like `type: constr('mytype') = 'mytype'`.